### ISSUE TITLE
board.has_kingside_castling_rights keeps returning False

### ISSUE THREAD

**Issue by pavolkacej** [niklasf/python-chess#810]
Run the following code:
```
board = chess.Board()
print(board.has_kingside_castling_rights(chess.WHITE))
print(board.has_queenside_castling_rights(chess.WHITE))
print(board.has_kingside_castling_rights(chess.BLACK))
print(board.has_queenside_castling_rights(chess.BLACK))
```
all the values are False. But the new board should start with True castling rights. Even if I empty the pieces between the king and the rook, the method keeps returning False value.
Please help resolving this.

**Comment by pavolkacej** [comment]
I would expect `has_kingside_castling_rights` method to check fen representation of the board (as fen contains the information, if the white/black king can castle), and return the value in this regard. Having own castling rights method, while fen has that information is a little bit ambiguous.
Wouldn't it be better to create

`set_kingside_castling_rights`and
`get_kingside_castling_rights`, which work only with boolean, instead of these bitwise operations?

**Comment by niklasf** [comment]
Thanks for reporting. I can't reproduce the issue:

```python
>>> import chess
>>> board = chess.Board()
>>> board.has_kingside_castling_rights(chess.WHITE)
True
>>> board.has_queenside_castling_rights(chess.WHITE)
True
>>> board.has_kingside_castling_rights(chess.BLACK)
True
>>> board.has_queenside_castling_rights(chess.BLACK)
True
```

What's your `chess.__version__`?

---

Unfortunately `board.set_kingside_castling_rights(chess.WHITE, True)` would be ambiguous in Chess960. With X-FEN, you can specify a file like`board.set_castling_fen("H")`.

**Comment by pavolkacej** [comment]
Hi @niklasf 

My mistake, sorry for the confusion. I was calling board.clear() in another function, that caused that board.castling_rights were set to False. I have learned, that they can be set again using `chess_squares.castling_rights |= chess.BB_CORNERS`, and they are updated correctly, based on the move_stack afterwards.

At least I learned something more about bitmasks, that may come handy in the future :)

Thank you for quick response, I don't think there is any further action needed.
Best regards
